airflow/www/views.py

```diff
@@ -2653,42 +2653,39 @@ class ConfigurationView(AirflowBaseView):
     )
     def conf(self):
         """Shows configuration."""
-        raw = request.args.get('raw') == "true"
         title = "Airflow Configuration"
-        subtitle = AIRFLOW_CONFIG
-        # Don't show config when expose_config variable is False in airflow config
+
         if conf.getboolean("webserver", "expose_config"):
-            with open(AIRFLOW_CONFIG) as file:
-                config = file.read()
             table = [
                 (section, key, value, source)
                 for section, parameters in conf.as_dict(True, True).items()
                 for key, (value, source) in parameters.items()
             ]
-        else:
-            config = (
-                "# Your Airflow administrator chose not to expose the "
-                "configuration, most likely for security reasons."
+
+            if request.args.get('raw') == "true":
+                return Response(response=json.dumps(table), status=200, mimetype="application/json")
```

Comment:
Is there a reason why to use json? Returning plain text allowed users do the copy past of configuration. I see not use case for json.

Reply:
If we're going to return json we should just content negotiation (i.e. Accept headers)

Reply:
To be consistent, I figured the `raw=true` argument should also return the rendered configuration. The airflow.cfg content was loaded as lines of text, but the rendered configuration is loaded as a list of tuples. So, I had to decide on an output format and figured JSON would make sense.

Thinking about it, the new REST API does exactly what we want (content negotiation, (https://github.com/apache/airflow/blob/master/airflow/api_connexion/endpoints/config_endpoint.py#L66-L87). We could also simply remove the `raw=true` argument and tell people to use the REST API, instead of maintaining this? I've never seen anybody using it?

Reply:
Went digging for possible usage, the `raw=true` code was first added 5 years ago: https://github.com/apache/airflow/commit/d12dc188d97f83d29bb8bba8b7b10c452f4c159a. Seems like "first iteration" was permanent after all😄

Shall I remove the `raw=true` argument and direct people to the REST API in UPDATING.md?

Reply:
Went ahead and removed `raw=true`

Reply:
If I correctly understand removing the `raw` option marks this PR automatically as Airflow 3.0, right? Should we do some type of depreciation warning first and remove it with 3.0?

I know this is probably rarely used feature but we decided to use semver.